### PR TITLE
feat(expr): Various trigonometric functions

### PR DESCRIPTION
**Pull request by CAJan93**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Implement the following math functions 

- `sin` / `cos` / `tan` / `cot` / `asin` / `acos` / `atan` / `atan2`

Related PR: https://github.com/risingwavelabs/risingwave/issues/8806 

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/common/src/types/decimal.rs [hunk 1]
```diff
@@ -499,6 +500,71 @@ impl Decimal {
         }
     }
 
+    #[must_use]
+    pub fn sin(&self) -> Self {
```
Comment (xiangjinwu) [inline src/common/src/types/decimal.rs:504]: FYI In PostgreSQL, trigonometric functions are only defined for double precision. Inputs of other types are [implicitly casted](https://github.com/risingwavelabs/risingwave/blob/c2decffad789a1e9361ab6c08634c8cc6aef603d/src/frontend/src/expr/type_inference/func.rs#L51) to double precision, and the output is always double precision.

Comment (CAJan93) [reply src/common/src/types/decimal.rs]: Thank you for pointing that out :)

File: src/workspace-hack/Cargo.toml [hunk 2]
```diff
@@ -18,191 +18,191 @@ publish = false
 
 ### BEGIN HAKARI SECTION
 [dependencies]
-ahash = { version = "0.8" }
-anyhow = { version = "1", features = ["backtrace"] }
+ahash = { version = "0.8", features = ["getrandom", "runtime-rng", "std"] }
```
Comment (CAJan93) [inline src/workspace-hack/Cargo.toml:21]: These changes were added after `risedev c`

**Comment by CAJan93** [comment]
Thank you for your feedback. I changed the tests slightly (see https://github.com/risingwavelabs/risingwave/pull/8838/commits/7d53b64d5badb9413f351de1397ba612381dcee8). These are mostly rounding errors. I assume those differences are ok?

**Pull request merged**